src/main/java/org/kohsuke/github/GitHub.java

```diff
@@ -205,6 +205,15 @@ public static GitHub connectAnonymously() throws IOException {
     }
 
     /**
+     * Connects to GitHub Enterprise anonymously.
```

Comment:
AFAIR Many of connect methods are useless and exist only for backward compatibility. Right logic should be use exist connects and builders. I see no reason for making separate methods or using anonymous in any env at all.